### PR TITLE
CM-962: What is the Difference Between MapControllers and MapControllerRoute in ASP.NET Core

### PR DESCRIPTION
**Pull request by emekaemego**

### CODE REVIEW

File: aspnetcore-features/MapControllersVsMapControllerRouteSln/MapControllersVsMapControllerRoute/Program.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+builder.Services.AddControllersWithViews();
```
Comment (fl0tschi) [inline aspnetcore-features/MapControllersVsMapControllerRouteSln/MapControllersVsMapControllerRoute/Program.cs:4]: `AddController()` instead

File: aspnetcore-features/MapControllersVsMapControllerRouteSln/MapControllersVsMapControllerRoute/Program.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+app.UseStaticFiles();
+
+app.UseRouting();
+
+app.UseAuthorization();
+// app.MapControllers();
```
Comment (fl0tschi) [inline aspnetcore-features/MapControllersVsMapControllerRouteSln/MapControllersVsMapControllerRoute/Program.cs:22]: Don't comment out code, let the code there, or if you don't need it remove it.

File: aspnetcore-features/MapControllersVsMapControllerRouteSln/MapControllersVsMapControllerRoute/Properties/launchSettings.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  "$schema": "http://json.schemastore.org/launchsettings.json",
+  "iisSettings": {
+    "windowsAuthentication": false,
+    "anonymousAuthentication": true,
+    "iisExpress": {
+      "applicationUrl": "http://localhost:6188",
+      "sslPort": 44303
+    }
+  },
+  "profiles": {
```
Comment (fl0tschi) [inline aspnetcore-features/MapControllersVsMapControllerRouteSln/MapControllersVsMapControllerRoute/Properties/launchSettings.json:11]: Only keep "https" profile. Rest should be removed.

File: aspnetcore-features/MapControllersVsMapControllerRouteSln/Tests/ConventionIntegrationTest.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+using Microsoft.AspNetCore.Mvc.Testing;
+
+namespace Tests;
+
+public class ConventionIntegrationTest: IClassFixture<WebApplicationFactory<Program>>
```
Comment (fl0tschi) [inline aspnetcore-features/MapControllersVsMapControllerRouteSln/Tests/ConventionIntegrationTest.cs:5]: Space between colon and text missing

File: aspnetcore-features/MapControllersVsMapControllerRouteSln/Tests/ConventionIntegrationTest.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    public ConventionIntegrationTest(WebApplicationFactory<Program> factory)
+    {
+        _factory = factory;
+    }
+    
+    [Theory]
+    [InlineData("/")]
+    [InlineData("/Home")]
+    [InlineData("/Home/Index")]
+    public async Task WhenMapControllerRouteRun_ThenSuccess(string url)
```
Comment (fl0tschi) [inline aspnetcore-features/MapControllersVsMapControllerRouteSln/Tests/ConventionIntegrationTest.cs:18]: We should test all controllers I would say.

File: aspnetcore-features/MapControllersVsMapControllerRouteSln/MapControllersVsMapControllerRoute/Controllers/HomeController.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
+using System.Diagnostics;
+using Microsoft.AspNetCore.Mvc;
+using MapControllersVsMapControllerRoute.Models;
+
+namespace MapControllersVsMapControllerRoute.Controllers;
+
+public class HomeController : Controller
```
Comment (fl0tschi) [inline aspnetcore-features/MapControllersVsMapControllerRouteSln/MapControllersVsMapControllerRoute/Controllers/HomeController.cs:7]: Don't use a view controller. It pollutes the PR with unnecessary files which are not necessary for the article itself. If we would use a similar one then the CustomersController or even re-use (if possible) that would be the best option.

**Review by fl0tschi — CHANGES_REQUESTED** [review]
Please remove the HomeController related files for the views. It would be better to use only simple API controller.

File: aspnetcore-features/MapControllersVsMapControllerRouteSln/Tests/ConventionIntegrationTest.cs (hunk 5 above)
Comment (emekaemego) [reply aspnetcore-features/MapControllersVsMapControllerRouteSln/Tests/ConventionIntegrationTest.cs:18]: The "Tests" project covers route testing for both controllers.

File: aspnetcore-features/MapControllersVsMapControllerRouteSln/MapControllersVsMapControllerRoute/Controllers/HomeController.cs (hunk 6 above)
Comment (emekaemego) [reply aspnetcore-features/MapControllersVsMapControllerRouteSln/MapControllersVsMapControllerRoute/Controllers/HomeController.cs:5]: The HomeController as a View controller will better convey the MapControllerRoute usage in the article. All the view files have been removed except for the Index.cshtml which is needed for the demonstration of the MapControllerRoute approach.

**Pull request closed without merging**